src/main/java/com/google/sps/servlets/utils/Statistics.java

```diff
@@ -57,9 +69,47 @@ public double getAverageScore() {
    * @param topNComments the number of highest magnitudes to retrieve
    */
   public Statistics(List<UserComment> userCommentList, int topNComments) {
+    wordsToIgnore = CommonWordsRetriever.getCommonWords();
     sentimentBucketList = categorizeToBucketList(userCommentList, topNComments);
     averageScore = getAverageValue(userCommentList, "score");
     averageMagnitude = getAverageValue(userCommentList, "Magnitude");
+    wordFrequencyMap = countWordFrequencyMap(userCommentList);
+  }
+
+  /**
+   * Convert given userCommentList into a word map: {word: frequency}
+   *
+   * @param userCommentList a list of userComment with all fields updated
+   * @return wordFrequencyMap to represent each word appearance time
+   */
+  private Map<String, Integer> countWordFrequencyMap(List<UserComment> userCommentList) {
+    // Flatten all user comment message into a list of words
+    Stream<String> allWordStream =
+        // Text extractor removes all HTML tags and returns only the text
+        userCommentList.stream()
+            .map(
+                comment ->
+                    new Source(comment.getCommentMsg())
+                        .getTextExtractor()
+                        .toString()
+                        .replaceAll("[^a-zA-Z0-9\\s]", "")
+                        .split("\\s+"))
+            .map(wordArray -> new ArrayList<>(Arrays.asList(wordArray)))
+            .flatMap(wordList -> wordList.stream())
+            .filter(word -> !wordsToIgnore.contains(word.toLowerCase()));
+    // Group and sum the appearances of each word
+    Map<String, Integer> wordPairMap =
+        allWordStream.collect(
+            Collectors.groupingBy(word -> word, Collectors.summingInt(word -> 1)));
```

Comment:
Optional:

I believe you should be able to change the `Collectors.summingInt(word -> 1)` to `Collectors.counting()`.

The only other change would be making it so that your map is declared as a `Map<String, Long>`.

Reply:
It does work with the change, but why would this be necessary? For the front end word cloud api, it needs to be <string, int>. We could change from long to int in the js file, but is it worth it

Reply:
Yeah it's fine in its current state, just offering another shorthand for summing.